### ISSUE TITLE
Moving out Shell/Globb

### ISSUE THREAD

**Issue by AbdealiJK** [coala-analyzer/coala#1869]
It would be nice to move out common things from coala into it's own package.

I was thinking:
 - StringProcessing
 - Shell (and a few context managers like `capture_stdout()` ?)
 - Globbing

Reason: I wanted windows support in another project and wanted `Shell` :P
(I don't want to maintain it there as @Makman2 already does such a good job :))

**Comment by rresol** [comment]
I would like to work on this issue.

**Comment by AbdealiJK** [comment]
@sils1297 Would like your opinion on this
On gitter @Makman2 said he was fine with it

We need to decide exactly what all needs to be moved. So adding the needs-design label.
@rresol I would recommend you not solve this yet

**Comment by sils1297** [comment]
I'm totally fine with that, it's great. Don't think it's difficulty low and we want an own issue for shell, stringprocessing and globbing. Maybe the latter two could go together in one lib.

**Comment by Makman2** [comment]
yeah a shell-lib and something like a pattern-matching-lib are good.
What about calling them `coashell` and `coamatch`? (dunno ... :P)

**Comment by sils1297** [comment]
I wouldn't prefix coa, they're generic

**Comment by Makman2** [comment]
:3

**Comment by sils1297** [comment]
@Makman2 will yo utake this? You'll likely end up maintaining them anyway, right?

**Comment by Makman2** [comment]
alright we need some repo names then:
- Globbing: `pyglob`, `globextensions`, `globext`, `globx` ?
- StringProcessing: `stringext`, `regexext`, `regext` `re-extensions`, `rextensions`?
- Shell: `shellext`, `shellextensions`?
(yeah I know everything with *"extension"*, but that were my first ideas...)

**Comment by AbdealiJK** [comment]
A few more:
StringProcessing -> `textlib`, `stringlib`, `texttools`
Shell -> `term`, `shlext`
Globbing -> `globber`, `globy`

**Comment by Makman2** [comment]
For stringprocessing I kinda like `rextensions`, because of `re`-module and `extensions` :D (I came also to `pyrex`, but that's already taken :3)
`shlext` is also cool.
And for globbing hmm... maybe we just call it `supernova` x)

**Comment by Makman2** [comment]
I wouldn't recommend to move out `Shell`, as there are only a few functions, they are more or less coala specific and the maintenance overhead for such a small package is not wort it imo. Recently I deleted some of the shell functions as they didn't work that well cross-platform.

**Comment by sils** [comment]
We should definitely consider improving an existent glob module from PyPI to do the job better.

**Comment by Makman2** [comment]
@sils you got some existing module?

**Comment by sils** [comment]
search pypi, we actually investigated this kind of thing once I think

**Comment by jayvdb** [comment]
I foubs globre to be very good, except no response ti one of my PRs.

https://github.com/metagriffin/globre/pulls

**Comment by jack17529** [comment]
@sils @Makman2 does this issue involves coding in Bash shell or only in python?

**Comment by Makman2** [comment]
just python.

**Comment by Asalle** [comment]
Should I move `Globbing` and `Shell` to coala-utils (the one on gitlab)? `StringProcessing` is already there.

**Comment by Makman2** [comment]
still needs discussion how and what components are moved, and maybe into a completely new package^^
I'm not sure what to do here^^ maybe you have some suggestions @Asalle ?

**Comment by Asalle** [comment]
This depends on what do you want to achieve. Do you want to move all generic things out of coala to reuse in other projects? Why wouldn't we just leave `Globbing` and `Shell` as they are?

**Comment by Makman2** [comment]
Those modules can maybe reused somewhere else, especially globbing could be interesting for others too. Those components rarely change in coala, and modularization is always good :+1:

**Comment by Makman2** [comment]
But we don't use them anywhere else^^

**Comment by Asalle** [comment]
ok, in such case we could move them out and use them as a dependency to coala
I can try and see how's going :)
btw, why do you use gitlab for coala-utils?

**Comment by Makman2** [comment]
GitLabs features match more our workflow, so we started hosting some repos there. However, migrating coala completely is some heavy task, so we haven't done this yet, it's not worth the effort right now ;)
